### PR TITLE
Make map stage final and refactor OneBoundedSetup to take a decider.

### LINKED ISSUE #19834: Replace all PushStages usages with GraphStages
- [x] Collect in progress by @agolubev
- [x] Drop
- [x] DropWhile in progress by @agolubev
- [x] Filter in progress by @miciek
- [x] LimitWeighted WIP by @agolubev
- [x] Log  in progress by @miciek
- [x] Map  in progress by @miciek
- [x] StartTimed in progress by @vans239,@naferx
- [x] StopTimed in progress by @vans239, @naferx
- [x] TakeWhile in progress by @vans239
- [x] TimedInterval in progress by @vans239, @naferx
- [x] UTF8Encoder WIP by @agolubev
- [x] PrepareForUserHandler WIP by @agolubev
- [x] CheckContentLengthTransformer in progress by @vans239 PR #19913
- [x] Take in progress by @trautonen 
- [x] captureTermination by @leachbj

### PR DESCRIPTION
**Pull request by miciek**
Resolves #19834

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

File: akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSupervisionSpec.scala [hunk 1]
```diff
@@ -62,8 +56,9 @@ class InterpreterSupervisionSpec extends StreamSpec with GraphInterpreterSpecKit
       lastEvents() should be(Set(Cancel, OnError(TE)))
     }
 
-    "resume when Map throws" in new OneBoundedSetup[Int](
-      new ResumingMap((x: Int) ⇒ if (x == 0) throw TE else x)
+    "resume when Map throws" in new OneBoundedSetupWithDecider[Int](
+      Supervision.resumingDecider,
+      new Map((x: Int) ⇒ if (x == 0) throw TE else x)
```
Comment (johanandren) [inline akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSupervisionSpec.scala]: Map is a case class so no need for `new`.

**Comment by johanandren** [comment]
LGTM

File: akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSupervisionSpec.scala (hunk 1 above)
Comment (miciek) [inline akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSupervisionSpec.scala]: OK. Done.

**Comment by patriknw** [comment]
LGTM

**Comment by drewhk** [comment]
Great thank you so much!

**Pull request merged**